test/hmap/llist_dscheck.ml

```diff
@@ -0,0 +1,242 @@
+let try_add ll elt = Llist.try_add ll elt ()
+
+module Sint = Set.Make (struct
+  type t = int
+
+  let compare = compare
```

Comment:
`Set.Make (Int)` would be preferable.  The `compare` here refers to the polymorphic compare, which is much slower than `Int.compare`.